salt/modules/mac_timezone.py

```diff
@@ -0,0 +1,436 @@
+        msg = 'Command failed: {0}'.format(ret['stderr'])
+        raise CommandExecutionError(msg)
+
+    return ret['stdout']
+
+
+def _parse_return(data):
+    '''
+    Parse a return in the format:
+    ``Time Zone: America/Denver``
+    to return only:
+    ``America/Denver``
+
+    Returns: The value portion of a return
+    '''
+
+    if ': ' in data:
+        return data.split(': ')[1]
+    if ':\n' in data:
+        return data.split(':\n')[1]
+    else:
+        return data
+
+
+def get_date():
+    '''
+    Displays the current date
+
+    Returns: the system date
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' timezone.get_date
+    '''
+    cmd = 'systemsetup -getdate'
+    ret = _execute_return_result(cmd)
+
+    return _parse_return(ret)
+
+
+def set_date(date):
+    '''
+    Set the current month, day, and year
+
+    :param str date: The date to set. Valid date formats are:
+    - %m:%d:%y
+    - %m:%d:%Y
+    - %m/%d/%y
+    - %m/%d/%Y
+
+    :return: True if successful, False if not
+    :rtype: bool
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' timezone.set_date 1/13/2016
+    '''
+    date_format = _get_date_time_format(date)
+    dt_obj = datetime.strptime(date, date_format)
+
+    cmd = 'systemsetup -setdate {0}'.format(dt_obj.strftime('%m:%d:%Y'))
+    _execute_return_success(cmd)
+
+    new_date = get_date()
+    date_format = _get_date_time_format(new_date)
+    new_dt_obj = datetime.strptime(new_date, date_format)
+
+    return dt_obj.strftime('%m:%d:%Y') == new_dt_obj.strftime('%m:%d:%Y')
+
+
+def get_time():
+    '''
+    Get the current system time.
+
+    :return: The current time in 24 hour format
+    :rtype: str
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' timezone.get_time
+    '''
+    cmd = 'systemsetup -gettime'
+    ret = _execute_return_result(cmd)
+
+    return _parse_return(ret)
+
+
+def set_time(time):
+    '''
+    Sets the current time. Must be in 24 hour format.
+
+    :param str time: The time to set in 24 hour format.
+    The value must be double quoted.
+
+    :return: True if successful, False if not
+    :rtype: bool
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' timezone.set_time '"17:34"'
+    '''
+    # time must be double quoted '"17:46"'
+    time_format = _get_date_time_format(time)
+    dt_obj = datetime.strptime(time, time_format)
+
+    cmd = 'systemsetup -settime {0}'.format(dt_obj.strftime('%H:%M:%S'))
+    _execute_return_success(cmd)
+
+    new_time = get_time()
+    time_format = _get_date_time_format(new_time)
+    new_dt_obj = datetime.strptime(new_time, time_format)
+
+    return dt_obj.strftime('%H:%M:%S') == new_dt_obj.strftime('%H:%M:%S')
+
+
+def get_zone():
+    '''
+    Displays the current time zone
+
+    :return: The current time zone
+    :rtype: str
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' timezone.get_zone
+    '''
+    cmd = 'systemsetup -gettimezone'
```

Comment:
There are a number of cases here where we're doing assignments and then passing them directly into a function. Could we just do `ret = _execute_return_result('systemsetup -gettimezone')` instead?